src/declarativewebcontainer.cpp

```diff
@@ -139,6 +135,10 @@ void DeclarativeWebContainer::setWebPage(DeclarativeWebPage *webPage)
         m_webPage = webPage;
 
         if (m_webPage) {
+            connect(m_webPage, SIGNAL(navigationHistoryChanged()), this, SIGNAL(canGoForwardChanged()));
+            connect(m_webPage, SIGNAL(navigationHistoryChanged()), this, SIGNAL(canGoBackChanged()));
+            connect(m_webPage, SIGNAL(urlChanged()), this, SIGNAL(urlChanged()));
+            connect(m_webPage, SIGNAL(titleChanged()), this, SIGNAL(titleChanged()));
```

Comment:
Why that would be a better place? I'd rather move all those `connect`'s from `activatePage()` to `setWebPage()`, because this way it's clearer why the connects should go after setting `m_webPage`.

Also the forward unconditional signaling seems to me easier to reason about than chained signaling.